.github/workflows/ci.yml

```diff
@@ -19,9 +19,7 @@ jobs:
           profile: minimal
```

Comment:
We're technically still using `actions-rs` to setup the toolchain in multiple places. The whole org is archived, I imagine that includes the toolchain repo (https://github.com/actions-rs/toolchain). Though that can be a separate PR since setup is a bit more complicated.

I assume this toolchain code wraps the rustup process, so that only matters if we're actually doing something other than stable. It's a bit annoying to repeat everywhere, so I can still see value in having an action for it if we start needing other toolchains, but currently I imagine you could even rip that out (maybe, I'm less sure of that, honestly - IDK when the containers for GitHub actions is updated to have a new rust version, so it's not clear to me that it's a no-op yet). Anyways, I am struggling to see the value of the original `actions-rs/cargo` repo.

Seems like the main thing we get from it is some support for recursing into rustup for cross-compilation, which is a feature we're not anyways using (yet): https://github.com/actions-rs/cargo/blob/master/src/main.ts

Aside from that, this is a pretty basic action that anyways ends up just calling `cargo ...`. I think we can rip it out and then if the need arises for cross-compilation we can re-evaluate what to do in the future.